### PR TITLE
Busthur

### PR DESCRIPTION
**Pull request by sueanyv**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+new Product('usb', 'img/usb.gif', 0, 0);
+new Product('water-can', 'img/water-can.jpg', 0, 0);
+new Product('wine-glass', 'img/wine-glass.jpg', 0, 0);
+
+// product constructor
+function Product(imgName, path, views, clicks) {
+  this.imgName = imgName;
+  this.path = path;
+  this.views = views;
+  this.click = clicks;
```
Comment (smitzter) [inline app.js]: this.click(s)?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  var data = {
+    labels: productImg.map(function(product) {
+      return product.imgName;
+    }),
+    datasets: [{
+      label: 'Clicks',
+      data: productImg.map(function(product) {
+        return product.click;
+      }),
+      backGroundColor: clickBackroundColors,
```
Comment (smitzter) [inline app.js]: clickBack(g)roundColors?

File: app.js (hunk 1 above)
Comment (smitzter) [inline app.js]: nevermind, it looks like you are using click throughout the rest.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  productImg.push(this);
+};
+
+function turnArrayIntoImages(product, img) {
+  var att = document.createAttribute('src');
+  att.value = product.path;
+  img.setAttributeNode(att);
+}
+
+function render() {
```
Comment (wrathofdon) [inline app.js]: you have a function within a function for no reason when both independent functions should be separate in the code, which might be creating variable scope issues.

File: app.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  var imgRandom = function() {
+    leftImg = Math.floor(Math.random() * productImg.length);
+    centerImg = Math.floor(Math.random() * productImg.length);
+    rightImg = Math.floor(Math.random() * productImg.length);
+    newImg = [];
+    newImg.push(leftImg, centerImg, rightImg);
+  };
+  imgRandom();
+  //checking for duplicate images
+  while ((leftImg === centerImg) || (centerImg === rightImg) || (leftImg === rightImg) || newImg[0] === previousImages[0] || newImg[0] === previousImages[1] || newImg[0] === previousImages[2] || newImg[1] === previousImages[0] || newImg[1] === previousImages[1] || newImg[1] === previousImages[2] || newImg[2] === previousImages[0] || newImg[2] === previousImages[1] || newImg[2] === previousImages[2]) {
```
Comment (wrathofdon) [inline app.js]: You can use the indexOf or includes methods to make this a lot more concise.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

**Pull request merged**